packages/remirror__extension-code-block/src/code-block-utils.ts

```diff
@@ -260,82 +261,108 @@ export function codeBlockToDOM(node: ProsemirrorNode, extra: ApplySchemaAttribut
   return ['pre', attributes, ['code', { [LANGUAGE_ATTRIBUTE]: language, style }, 0]];
 }
 
-interface FormatCodeBlockFactoryProps
-  extends NodeTypeProps,
-    Required<Pick<CodeBlockOptions, 'formatter' | 'defaultLanguage'>> {}
+type FormatCodeProps = NodeTypeProps &
+  Required<Pick<CodeBlockOptions, 'formatter' | 'defaultLanguage'>> &
+  Partial<PosProps> & { state: EditorState };
+
+export interface FormatCodeResult {
+  /**
+   * Formatted code
+   */
+  formatted: string;
+
+  /**
+   * The original range that should be replaced in the code block
+   */
+  range: { from: number; to: number };
+
+  /**
+   * Updated selection coordinates for the formatted code
+   */
+  selection: { anchor: number; head?: number };
+}
 
 /**
- * A factory for creating a command which can format a selected codeBlock (or
- * one located at the provided position).
+ * Format the contents of a selected codeBlock or one located at the provided
+ * position.
  */
-export function formatCodeBlockFactory(props: FormatCodeBlockFactoryProps) {
-  return ({ pos }: Partial<PosProps> = object()): CommandFunction =>
-    ({ tr, dispatch }) => {
-      const { type, formatter, defaultLanguage: fallback } = props;
+export async function formatCode(props: FormatCodeProps): Promise<FormatCodeResult> {
+  const { type, formatter, defaultLanguage: fallback, pos, state } = props;
+  const { selection, doc } = state;
```

Comment:
Again, following on the the arguments vs command function props mentioned above

You will have `tr` available on props now, and we should that instead of `state`

```ts
  const { type, formatter, defaultLanguage: fallback, pos, tr } = props;
  const { selection, doc } = tr;
```